Source/WebKit/UIProcess/Extensions/WebExtension.h

```diff
@@ -345,48 +351,53 @@ class WebExtension : public API::ObjectImpl<API::Object::Type::WebExtension>, pu
 
     MatchPatternSet m_externallyConnectableMatchPatterns;
 
+#if PLATFORM(COCOA)
     RetainPtr<NSBundle> m_bundle;
     mutable RetainPtr<SecStaticCodeRef> m_bundleStaticCode;
-    RetainPtr<NSURL> m_resourceBaseURL;
-    RetainPtr<NSDictionary> m_manifest;
-    RetainPtr<NSMutableDictionary> m_resources;
+#endif
+
+    URL m_resourceBaseURL;
+    RefPtr<JSON::Value> m_manifest;
```

Comment:
A manifest is required for a WebExtension (logically speaking), so if its possible to make this a `Ref` instead so its never `nullptr` that would be ideal.

Reply:
logically yes, but parsing the manifest can still fail. the manifest() function returns nullptr a few times for if the manifest fails to parse. I'm not sure what to do instead there

Reply:
i see you commented on this below, sorry